### DISCUSSION TITLE
Does sea-orm support AsChangeset feature

### DISCUSSION THREAD

**Discussion by baoyachi** [SeaQL/sea-orm#831] [Q&A]
link diesel AsChangeset:https://docs.rs/diesel/latest/diesel/query_builder/trait.AsChangeset.html#

**Comment by billy1624** [comment]
Hey @baoyachi, welcome! I think it's similar to `ActiveModel` in SeaORM
- https://www.sea-ql.org/SeaORM/docs/basic-crud/insert#set-activemodel-from-json-value

**Reply by baoyachi** [reply]
Diesel support some fields can be modified, and the existing results will not be changed when other fields are set to none.

E.g:
* original struct
```
pub struct Model {
    pub id: i32,
    pub name: String,
    pub addr: String,
    pub number: String,
}
```
* Now I want update `addr` filed, then I can do this:

```
pub struct UpdateModel {
    pub name:Option<String>,
    pub addr: Option<String>,
    pub number: Option<String>,
}
```,
exec diesel code:

```rust
let update_model = UpdateModel{
   name:None,
  addr:String::from("center street");
 number:None,
};

let updated_row = diesel::update(model.table)
    .filter(model::id.eq(1))
    .set(update_model)
    .get_result(&connection);
```

Does sea-orm support a part filed updated?
